src/model/censusdata/censusdata.js

```diff
@@ -33,6 +38,50 @@ export function reset() {
   categoryCodeLookup = {};
 }
 
+export async function fetchAllDataForGeography(censusDataService, geographyCode) {
+  dataService = censusDataService;
+  const data = await dataService.fetchAllDataForGeography(geographyCode);
+  selectedGeographyData.set(data);
+}
+
+export async function fetchSelectedDataForGeoType(censusDataService, geoType, categories) {
+  dataService = censusDataService;
+  const data = await dataService.fetchSelectedDataForGeographyType(geoType, categories);
+  dataByGeography.set(data);
+}
+
+export async function fetchSelectedDataForGeographies(censusDataService, geoCodes, catCodes) {
+  dataService = censusDataService;
+  const data = await dataService.fetchSelectedDataForGeographies(geoCodes, catCodes);
+  dataByGeography.set(data);
+}
+
+function filtermapBBoxCodes(dataByGeography, mapBBoxCodes) {
+  if (dataByGeography.length == 0) {
+    return mapBBoxCodes;
+  }
+  return mapBBoxCodes.filter((item) => !dataByGeography.has(item));
+}
+
+export async function fetchSelectedDataForNewBoundingBoxGeographies(censusDataService, catCodes) {
+  dataService = censusDataService;
+  const geoCodes = filtermapBBoxCodes(get(dataByGeography), get(mapBBoxCodes));
+  const data = await dataService.fetchSelectedDataForGeographies(geoCodes, catCodes);
+  data.forEach((data, key) => {
+    const catCode = Object.keys(data);
+    get(dataByGeography).set(key, { [catCode]: data[catCode] });
+  });
+  //temporarily sets store to true to so components can listen for new data
+  newDataByGeography.set(true);
+  newDataByGeography.set(false);
+}
+
+export async function fetchSelectedDataForWholeBoundingBox(censusDataService, geoTypes, catCodes, bBox) {
+  dataService = censusDataService;
+  const data = await dataService.fetchSelectedDataForBoundingBox(geoTypes, catCodes, bBox);
+  dataByGeography.set(data);
```

Comment:
why does newDataByGeography not need to be toggled here?

Reply:
good question, and I'm not really sure what the answer is.  Maybe when you (re)set the whole Map, under the hood Svelte treats that as a reassignment?  Unlike when you set the Map one key value pair at a time, as in the for each above

Reply:
Reading this on my phone, so probably missing something, but why is the routine different for loading data from a bbkx request vs from a named geographies request?

Reply:
The named geographies request generates a URL containing all the geography codes, whereas the bbox request just uses the lat/long coordinates and lets the back end work out what the codes are

Reply:
but the data they receive should have the same structure, right? Just wondering why `fetchSelectedDataForNewBoundingBoxGeographies` iterates through the data and calls `dataByGeography.set` on each key, value pair, whereas `fetchSelectedDataForNewBoundingBoxGeographies` just calls `dataByGeography.set` on the data it gets back, without iterating through it?

Reply:
ah I see. The logic was that if you're using `fetchSelectedDataForWholeBoundingBox`, it would make sense to write over any cached data in dataByGeography, whereas the use case for `fetchSelectedDataForNewBoundingBoxGeographies` is when the user is moving around the map, and in that situation we would generally want to leave the existing data as is and add the new geographies to it.  Which requires us to set the new geographies key by key, in the forEach loop.

Reply:
I think all these functions should have an optional parameter that overwrites the cache if true - I can definitely imagine scenarios in which you want bbox data but you don't want to invalidate the entire cache?

Reply:
as discussed offline - this will be address in a future PR.